plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/TokenFetchingMetastoreClientFactory.java

```diff
@@ -58,6 +66,12 @@ public TokenFetchingMetastoreClientFactory(
                         .maximumSize(thriftConfig.getDelegationTokenCacheMaximumSize()),
                 CacheLoader.from(this::loadDelegationToken));
         this.refreshPeriod = Duration.ofMinutes(1).toNanos();
+        retryPolicy = RetryPolicy.builder()
+                .withMaxDuration(java.time.Duration.of(30, SECONDS))
+                .withMaxAttempts(thriftConfig.getMaxRetries() + 1)
+                .withBackoff(thriftConfig.getMinBackoffDelay().toMillis(), thriftConfig.getMaxBackoffDelay().toMillis(), MILLIS, thriftConfig.getBackoffScaleFactor())
+                .abortOn(throwable -> Throwables.getCausalChain(throwable).stream().anyMatch(TrinoException.class::isInstance))
```

Comment:
Did we figure which exceptions are permanent or temporary?
Is `MetaException` permanent?
Can `TException` be temporary?

Would `getDelegationToken` throw `TrinoExceptipon` at all (I don't think it does)?

Reply:
yes, we do not need to use `TrinoException`,

we can Abort on `TException` like it is followed for other operation's retry like `createDatabase` at the Metastore side throws three [exceptions](https://github.com/apache/hive/blob/master/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java#L1161-L1162) and Trino side we abort for these three [exceptions](https://github.com/trinodb/trino/blob/master/plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftHiveMetastore.java#L763) only.

Reply:
I though that maybe we can narrow it more, but it's fine for me